File: 400/412 - Fizz Buzz.py
class Solution(object):
    def fizzBuzz(self, n):
        """
        :type n: int
        :rtype: List[str]
        """
        l = []
        for i in range(n):
            i += 1
            if not (i % 15):
                l.append('FizzBuzz')
            elif not (i % 5):
                l.append('Buzz')
            elif not (i % 3):
                l.append('Fizz')
            else:
                l.append(str(i))
        return l

File: csharp/412.fizz-buzz.cs
/*
 * @lc app=leetcode id=412 lang=csharp
 *
 * [412] Fizz Buzz
 */

using System.Collections.Generic;
using System.Linq;

public class Solution {
    public IList<string> FizzBuzz(int n) {
        return new List<string>(Enumerable.Range(1, n).Select(x => {
            bool three = x % 3 == 0, five = x % 5 == 0;
            return three ?
                (five ? "FizzBuzz" : "Fizz") :
                (five ? "Buzz" : x.ToString());
        }));
    }
}

